### PR TITLE
[HIGH] add reject invoice dialog

### PR DESCRIPTION
**Pull request by mugnivenko**
https://app.clickup.com/t/1mf48qy

### CODE REVIEW

File: components/Pages/Payruns/ApproveDeclineModal/index.js [hunk 1]
```diff
@@ -68,9 +68,13 @@ const ApproveDeclineModal = ({ openedModal, setOpenedModal, update, payRunId })
...
   const handleReject = async (notes) => {
     try {
-      await rejectPayRun(payRunId, notes);
-      pushNotification(`Invoice status changed`, 'success');
-      update();
+      if (request) {
+        await request(notes);
+      } else {
+        await rejectPayRun(payRunId, notes);
+        pushNotification(`Pay run status changed`, 'success');
```
Comment (akeamwoodsND) [inline components/Pages/Payruns/ApproveDeclineModal/index.js:75]: if you're using this for invoices as well then you need to change pay run to a prop

File: components/Pages/Payruns/SinglePayRun/PayRunItem.js [hunk 2]
```diff
@@ -101,11 +132,19 @@ export const PayRunItem = ({
         <HoldPaymentDialog
           invoiceId={invoiceId}
           payRunId={payRunId}
-          isOpen={invoiceId}
           update={updateData}
-          setIsOpened={() => setInvoiceId('')}
+          isOpen={openedModal === 'Hold'}
+          setIsOpened={closeModal}
```
Comment (akeamwoodsND) [inline components/Pages/Payruns/SinglePayRun/PayRunItem.js:137]: setIsOpened=closeModal ?? rename closeModal

File: components/Pages/Payruns/SinglePayRun/PayRunItem.js [hunk 3]
```diff
@@ -101,11 +132,19 @@ export const PayRunItem = ({
...
         />
       )}
+      <ApproveDeclineModal
+        setOpenedModal={closeModal}
+        openedModal={openedModal !== 'Hold' && openedModal}
+        payRunId={payRunId}
+        title="Invoice"
+        request={rejectInvoiceStatus}
+        update={update}
```
Comment (akeamwoodsND) [inline components/Pages/Payruns/SinglePayRun/PayRunItem.js:146]: rename update prop to something more descriptive

File: pages/payruns/[guid]/index.js [hunk 4]
```diff
@@ -74,6 +75,9 @@ const SinglePayRun = () => {
     update();
     updateInsight();
   };
+
+  const isLoading = invoiceLoading || insightsIsLoading || loading;
```
Comment (akeamwoodsND) [inline pages/payruns/[guid]/index.js:79]: nope we dont want this logic being passed in

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]

File: components/Pages/Payruns/ApproveDeclineModal/index.js (hunk 1 above)
Comment (mugnivenko) [reply components/Pages/Payruns/ApproveDeclineModal/index.js]: added rejectRequest function

File: components/Pages/Payruns/SinglePayRun/PayRunItem.js (hunk 2 above)
Comment (mugnivenko) [reply components/Pages/Payruns/SinglePayRun/PayRunItem.js]: renamed components prop to closeModal

File: components/Pages/Payruns/SinglePayRun/PayRunItem.js (hunk 3 above)
Comment (mugnivenko) [reply components/Pages/Payruns/SinglePayRun/PayRunItem.js]: renamed update to isActivePayRun

File: pages/payruns/[guid]/index.js (hunk 4 above)
Comment (mugnivenko) [reply pages/payruns/[guid]/index.js]: removed loading

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
change 'decline invoice' to 'reject invoice' where it is used.

also change success message to 'Invoice has been rejected'

**Pull request merged**